rotkehlchen/db/search_assets.py

```diff
@@ -0,0 +1,119 @@
+                continue
+
+            substr_in_name = substr_in_symbol = None
+            # default to a value greater than the threshold to avoid false-positives
+            # and prevent raising `UnboundLocalError` when name/symbol is None.
+            lev_dist_name = lev_dist_symbol = 6
+            if entry[1] is not None:
+                name_casefold = entry[1].casefold()
+                lev_dist_name = levenshtein(substring_search, name_casefold)
+                substr_in_name = substring_search in name_casefold
+
+            if entry[2] is not None:
+                symbol_casefold = entry[2].casefold()
+                lev_dist_symbol = levenshtein(substring_search, symbol_casefold)
+                substr_in_symbol = substring_search in symbol_casefold
+
+            if not substr_in_name and not substr_in_symbol:
+                continue
+
+            lev_dist_min = min(lev_dist_name, lev_dist_symbol)
+            if treat_eth2_as_eth is True and entry[0] in (A_ETH.identifier, A_ETH2.identifier):  # noqa:E501
+                if found_eth is False:
+                    search_result.append({
+                        'identifier': resolved_eth.identifier,
+                        'name': resolved_eth.name,
+                        'symbol': resolved_eth.symbol,
+                        'asset_type': 'own chain',
+                    })
+                    levenshtein_distances.append(lev_dist_min)
+                    found_eth = True
+                continue
+
+            entry_info = {
+                'identifier': entry[0],
+                'name': entry[1],
+                'symbol': entry[2],
+                'asset_type': AssetType.deserialize_from_db(entry[4]).serialize(),
+            }
+            if entry[3] is not None:
+                entry_info['evm_chain'] = ChainID.deserialize_from_db(entry[3]).serialize()
+            if entry[5] is not None:
+                entry_info['custom_asset_type'] = entry[5]
+
+            search_result.append(entry_info)
+            levenshtein_distances.append(lev_dist_min)
+
+        if search_nfts is True:
+            cursor.execute('SELECT identifier, name, collection_name FROM nfts')
+            for entry in cursor:
+                # if an nft does not have a name and collectio name, skip.
+                if entry[1] is None and entry[2] is None:
+                    continue
+
+                substr_in_name = substr_in_collection_name = None
+                # default to a value greater than the threshold to avoid false-positives
+                # and prevent raising `UnboundLocalError` when name/collection name is None.
+                lev_dist_name = lev_dist_collection_name = 6
+                if entry[1] is not None:
+                    name_casefold = entry[1].casefold()
+                    lev_dist_name = levenshtein(substring_search, name_casefold)
+                    substr_in_name = substring_search in name_casefold
```

Comment:
can't you add this and the constraint in the collection name with a `COLLATE NOCASE` and a `OR`? This will avoid searching in all the NFTs

Reply:
I guess no since this is for the levenshtein search that happens in python, not in sqlite. (if I understood correctly what you meant)

Reply:
(or at least this is how levenshtein search works for assets, so I would say that is better to keep to consistent)

Reply:
`substring_search in name_casefold` is a check that doesn't require anything from the lev search

Reply:
Yeah, but I would say that it is something that it better to change when touching levenshtein search? It was designed like this so maybe there was a reason? Idk

Reply:
When are we going to touch this logic? Have you checked doing it in SQL and moving this condition? Now that I check it more it makes less sense as how is now